apps/extension/src/App/Accounts/AddAccount.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button, ButtonVariant, Input, InputVariants } from "@anoma/components";
+
+import { ExtensionRequester } from "extension";
+import { Ports } from "router";
+import { DerivedAccount, DeriveAccountMsg } from "background/keyring";
+import {
+  AddAccountContainer,
+  AddAccountForm,
+  Bip44PathContainer,
+  Bip44PathDelimiter,
+  Bip44Input,
+  Bip44Path,
+  Bip44Error,
+  ButtonsContainer,
+  FormError,
+  FormStatus,
+  InputContainer,
+  Label,
+} from "./AddAccount.components";
+
+type Props = {
+  accounts: DerivedAccount[];
+  requester: ExtensionRequester;
+  setAccounts: (accounts: DerivedAccount[]) => void;
+};
+
+const validateAccount = (
+  newAccount: { account: number; change: number; index: number },
+  accounts: DerivedAccount[]
+): boolean => {
+  const newPath = [
+    newAccount.account,
+    newAccount.change,
+    newAccount.index,
+  ].join("/");
+  let isValid = true;
+  accounts.forEach((a: DerivedAccount) => {
+    const { bip44Path } = a;
+    const { account, change, index } = bip44Path;
+    const existingPath = [account, change, index].join("/");
+
+    if (newPath === existingPath) {
+      isValid = false;
+    }
+  });
+
+  return isValid;
+};
+
+const findNextIndex = (accounts: DerivedAccount[]): number => {
+  let maxIndex = 0;
+  accounts.forEach((account) => {
+    const { index } = account.bip44Path;
+    if (index > maxIndex) {
+      maxIndex = index;
+    }
+  });
+
+  return maxIndex + 1;
+};
+
+enum Status {
```

Comment:
nit: I feel that if we do state management in React components, we might like to do some small util/share something like this type. We only now have this 3 times, but likely will have it several times soon.

Reply:
I agree with this, but would first like to refer to the subsequent PRs that are out, as some of this has been moved around or removed. When using the `ExtensionRequester` `sendMessage` function, we should have one set of states that it could be in, so I definitely agree these should be consolidated.

That said, I don't think there is a good use case for doing state-management outside of what React provides, as it these are still very simple. While it _may_ grow somewhat in the future, I still think it will be very manageable with only using `react-router` and local state (or context for any globals)